plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,114 @@
-import komand
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+
 from .schema import ConnectionSchema, Input
-from komand.exceptions import ConnectionTestException
 
 # Custom imports below
 import requests
+from typing import Optional
+import json
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.job_id = None
 
-    def connect(self, params):
-        self.logger.info("Connect: Connecting..")
+    def connect(self, params={}):
+        self.logger.info("Connect: Connecting...")
         self.host = params.get(Input.URL)
         self.token = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_enriched_event(
+        self, criteria: dict, exclusions: dict = None, time_range: dict = None
+    ) -> Optional[dict]:
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs",
+            json_data={"criteria": criteria, "exclusions": exclusions, "time_range": {"window": time_range}},
+        )
+        return response.get("job_id")
+
+    def get_enriched_event_status(self, job_id: str = None) -> bool:
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v1/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}",
+            json_data={"cb_job_id": job_id},
+        )
+        contacted = response.get("contacted")
+        completed = response.get("completed")
+        if contacted and completed and (contacted == completed):
```

Comment:
```suggestion
        return contacted and completed and (contacted == completed)
```
[This is the way.](https://tenor.com/view/mando-way-this-is-the-way-mandalorian-star-wars-gif-18467370)

Reply:
This has been updated.